docs/tutorials/client-server.md

```diff
@@ -0,0 +1,230 @@
+
+ZeroMQ provides a number of socket types that could be used to achieve something approaching this architecture. The @router-dealer socket pair acting as Server / Client has a couple of advantages for this design: 
+- Routers assign a unique address for each connected client allowing clients in turn to be addressed individually
+- Messages can be passed between Router / Dealer sockets without the requirement that a reply is received before the next message is sent, as is the case with the Request / Response socket pair.
+
+## Basic client
+To begin with, we’ll create a simple client that sends basic messages on a network. In a new Bonsai project:
+- Add a [`Dealer`](xref:Bonsai.ZeroMQ.Dealer) operator. 
+- In the `ConnectionString` property, set `Address`: localhost:5557, `Action`: Connect, `Protocol`: TCP.
+
+> [!Note]
+> In Bonsai.ZeroMQ, the [`Dealer`](xref:Bonsai.ZeroMQ.Dealer) can have two functions based on its inputs. On its own, as above, the [`Dealer`](xref:Bonsai.ZeroMQ.Dealer) operator creates a Dealer socket that listens for messages on the specified network. With the properties specified, we are asking our [`Dealer`](xref:Bonsai.ZeroMQ.Dealer) to listen for messages on the local machine on port 5557 using the TCP protocol. We use the ‘Connect’ argument for the `SocketConnection` property to tell the dealer that it will connect to a static part of the network with a known IP address, in this case the server which we will implement later.
+
+If we add inputs to the [`Dealer`](xref:Bonsai.ZeroMQ.Dealer), it will act as both a sender and receiver of messages on the network. 
+- Before the [`Dealer`](xref:Bonsai.ZeroMQ.Dealer) operator add a [`KeyDown`](xref:Bonsai.Windows.Input.KeyDown). 
+- Add a [`String`](xref:Bonsai.Expressions.StringProperty) operator in sequence as input to the [`Dealer`](xref:Bonsai.ZeroMQ.Dealer).
+
+:::workflow
+![Basic Dealer input](~/workflows/dealer-basic-input.bonsai)
+:::
+
+- In the operator properties, set the [`KeyDown`](xref:Bonsai.Windows.Input.KeyDown) `Filter` to the ‘1’ key.
+- Set the [`String`](xref:Bonsai.Expressions.StringProperty) `Value` to ‘Client1’. 
+
+If we run the Bonsai project now, the [`Dealer`](xref:Bonsai.ZeroMQ.Dealer) will continue listening for incoming messages on the network, but every time the ‘1’ key is pressed a message containing the string ‘Client1’ will be sent from the socket.
+
+- Copy and paste this client structure a couple of times and change the [`KeyDown`](xref:Bonsai.Windows.Input.KeyDown) and [`String`](xref:Bonsai.Expressions.StringProperty) properties accordingly on each (2, ‘Client2’; 3, ‘Client3’) so that we have 3 total clients that send messages according to different key presses.
```

Comment:
```suggestion
- Copy and paste this client structure a couple of times and change the [`KeyDown`](xref:Bonsai.Windows.Input.KeyDown) and [`String`](xref:Bonsai.Expressions.StringProperty) properties accordingly on each (2, `Client2`; 3, `Client3`) so that we have a total of 3 clients which send messages according to different key presses.
```